ApplicationLibCode/ProjectDataModel/WellLog/RimRftTools.cpp

```diff
@@ -206,20 +203,43 @@ std::vector<double> RimRftTools::segmentStartMdValues( RifReaderRftInterface*
                                                      const QString&            wellName,
                                                      const QDateTime&          dateTime,
                                                      int                       segmentBranchIndex,
                                                      RiaDefines::RftBranchType segmentBranchType )
 {
+    if ( !readerRft ) return {};
+
     std::vector<double> values;
+    auto                resultAddress = RifEclipseRftAddress::createBranchSegmentAddress( wellName,
+                                                                           dateTime,
+                                                                           RiaDefines::segmentEndDepthResultName(),
+                                                                           segmentBranchIndex,
+                                                                           segmentBranchType );
+    readerRft->values( resultAddress, &values );
 
-    auto resultNameSeglenst = RifEclipseRftAddress::createBranchSegmentAddress( wellName,
-                                                                                dateTime,
-                                                                                RiaDefines::segmentEndDepthResultName(),
-                                                                                segmentBranchIndex,
-                                                                                segmentBranchType );
-    readerRft->values( resultNameSeglenst, &values );
+    return values;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+std::vector<double> RimRftTools::segmentConnectionMdValues( RifReaderOpmRft*          readerRft,
+                                                            const QString&            wellName,
+                                                            const QDateTime&          dateTime,
+                                                            int                       segmentBranchIndex,
+                                                            RiaDefines::RftBranchType segmentBranchType )
+{
+    if ( !readerRft ) return {};
+
+    std::vector<double> values;
+    auto                resultAddress = RifEclipseRftAddress::createBranchSegmentAddress( wellName,
+                                                                           dateTime,
+                                                                           RiaDefines::segmentConnectionMeasuredDepthResultName(),
+                                                                           segmentBranchIndex,
+                                                                           segmentBranchType );
+    readerRft->values( resultAddress, &values );
```

Comment:
Would be nice RftReaderInterface::values() would create the result array. Not really related to this PR, though..

Reply:
Agree, created separate issue https://github.com/OPM/ResInsight/issues/11712